src/lightning/pytorch/strategies/fsdp.py

```diff
@@ -168,23 +179,12 @@ def __init__(
             activation_checkpointing, activation_checkpointing_policy
         )
 
-    def lightning_module_state_dict(self) -> Dict[str, Any]:
-        """Gathers the full state dict by unsharding all the parameters.
-
-        To avoid OOM, the returned parameters will only be returned on rank 0 and on CPU. All other ranks get an empty
-        dict.
-
-        """
-        from torch.distributed.fsdp import FullStateDictConfig, FullyShardedDataParallel, StateDictType
-
-        assert self.model is not None
-
-        with FullyShardedDataParallel.state_dict_type(
-            module=self.model,
-            state_dict_type=StateDictType.FULL_STATE_DICT,
-            state_dict_config=FullStateDictConfig(offload_to_cpu=(self.world_size > 1), rank0_only=True),
```

Comment:
@carmocca See here, the code was here before, I'm merely just refactoring it so that it uses the existing logic from Fabric. The condition `offload_to_cpu=(self.world_size > 1)` is not new to this PR, it was there before.

Reply:
Ah I missed this. Thanks for the context